x/auth/client/cli/validate_sigs.go

```diff
@@ -89,40 +98,63 @@ func printAndValidateSigs(
 		// Validate the actual signature over the transaction bytes since we can
 		// reach out to a full node to query accounts.
 		if !offline && success {
-			acc, err := types.NewAccountRetriever(authclient.Codec).GetAccount(clientCtx, sigAddr)
+			accNum, accSeq, err := clientCtx.AccountRetriever.GetAccountNumberSequence(clientCtx, sigAddr)
 			if err != nil {
 				cmd.Printf("failed to get account: %s\n", sigAddr)
 				return false
 			}
 
-			sigBytes := types.StdSignBytes(
-				chainID, acc.GetAccountNumber(), acc.GetSequence(),
-				stdTx.Fee, stdTx.GetMsgs(), stdTx.GetMemo(),
-			)
-
-			if ok := sig.GetPubKey().VerifyBytes(sigBytes, sig.Signature); !ok {
-				sigSanity = "ERROR: signature invalid"
-				success = false
+			signingData := authsigning.SignerData{
+				ChainID:         chainID,
+				AccountNumber:   accNum,
+				AccountSequence: accSeq,
 			}
-		}
-
-		multiPK, ok := sig.GetPubKey().(multisig.PubKeyMultisigThreshold)
-		if ok {
-			var multiSig multisig.AminoMultisignature
-			clientCtx.Codec.MustUnmarshalBinaryBare(sig.Signature, &multiSig)
 
-			var b strings.Builder
-			b.WriteString("\n  MultiSig Signatures:\n")
+			switch data := sig.Data.(type) {
```

Comment:
@aaronc can't we use VerifySignature from here directly https://github.com/cosmos/cosmos-sdk/blob/c0e48e1c43f7054aea1a58aaceb17d354ccf98b5/x/auth/signing/verify.go#L15